src/util.ts

```diff
@@ -381,6 +382,11 @@ class Util {
         const seconds = allocate(1000);
         return `${hours > 0 ? hours + 'h' : ''}${minutes > 0 ? minutes + 'm' : ''}${seconds > 0 ? seconds + 's' : ''}${ms > 0 ? ms + 'ms' : ''}`;
     }
+
+    public normalizeAdapterOptions(options: AdapterOptions) {
```

Comment:
we can't default an ip but is there anything we should do here? Like throw an error? I know we "should" catch it earlier but wouldn't be bad to do here would it?

Reply:
Nah. This is mostly just to centralize the default ports. If we want to catch a missing host, that should be done at the vscode level or in the initialize function of the session.